stdlib/scanf.mli

```diff
@@ -310,14 +310,19 @@ val bscanf_opt : Scanning.in_channel -> ('a, 'b, 'c, 'd) scanner_opt
 
     The conversion characters and their meanings are:
 
-    - [d]: reads an optionally signed decimal integer ([0-9]+).
+    - [d]: reads an optionally signed decimal integer [[+-]?[0-9][0-9_]*]
+      (accepting underscores after the first digit is reminiscent of OCaml
+       lexical conventions).
```

Comment:
Since the exact regexp requires some time to read, I would propose to de-emphasize it:
```
reads an optionally signed decimal integer using OCaml lexical conventions;
[59],  [1_000], [+1], [-1] are all accepted like any number of the form [[+-]?[0-9][0-9_]*].
```